 #ADAPTIVE-FILTER Parameters
 adaptive_filter:

  # Filter settings
  enableFilter: true
  enableImu: true
  enableWheel: true
  enableLidar: true
  enableVisual: true
  filterFreq: "l"
  freq: 200.0

  # input filter
  alpha_lidar: 0.99
  alpha_visual: 0.999

  # gains  
  lidarG: 75
  wheelG: 0.5
  visualG: 0.1
  imuG: 100

  # Wheel odometry covariance adaptive positive constants
  gamma_vx: 0.05
  gamma_omegaz: 0.01
  delta_vx: 0.0001
  delta_omegaz: 0.00001

  # visual odometry covariance
  minIntensity: 0.0
  maxIntensity: 1.0

  # type of adaptative function
  lidar_type_func: 2                        # 0: linear; 1: exponential; 2: original message
  visual_type_func: 2                       # 0: linear; 1: exponential; 2: original message
  wheel_type_func: 1                        # 0: our function; 1: original message

  # type of cameras
  camera_type: 1                      # 1: tracking; 2: depth

